### ISSUE TITLE
Cannot compile

### ISSUE THREAD

**Issue by morispi** [francesco-peverelli/BMAN-GPU#1]
Hi,

I'm trying to compile the whole CONSENT-GPU pipeline to try it out, but I can't seem to be able do to compile the BMAN-GPU repo.

Here's what I get:

```
/gpfs2/soft/gcc-6.4.0/include/c++/6.4.0/tuple:495:244: error: wrong number of template arguments (4, should be 2)
       return  __and_<__not_<is_same<tuple<_Elements...>,
                                                                                                                                                                                                                                                    ^    
/gpfs2/soft/gcc-6.4.0/include/c++/6.4.0/type_traits:1558:8: note: provided for ‘template<class _From, class _To> struct std::is_convertible’
     struct is_convertible
        ^~~~~~~~~~~~~~
/gpfs2/soft/gcc-6.4.0/include/c++/6.4.0/tuple:502:1: error: body of constexpr function ‘static constexpr bool std::_TC<<anonymous>, _Elements>::_NonNestedTuple() [with _SrcTuple = std::tuple<void (*)(int, poa_gpu_utils::SyncMultitaskConcurrencyManager<std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >*), int, poa_gpu_utils::SyncMultitaskConcurrencyManager<std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >*>&&; bool <anonymous> = true; _Elements = {void (*)(int, poa_gpu_utils::SyncMultitaskConcurrencyManager<std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >*), int, poa_gpu_utils::SyncMultitaskConcurrencyManager<std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >*}]’ not a return-statement
     }
 ^
make: *** [bmean.o] Error 1
```

Do you know what might be causing the error?

Best,
P

**Comment by francesco-peverelli** [comment]
Hi,

I have pushed some fixes to the CONSENT_GPU repo, now you should be able to pull, together with the BMAN-GPU submodule and run install.sh  for the whole CONSENT-GPU pipeline.
I am using gcc 8.3.0 and CUDA 10.2. I have never encountered your issue, if you can try to pull a clean repo and run install with the same gcc version. I will run some test with earlier gcc versions in the next few days

**Comment by morispi** [comment]
Hi,

I think that may be the reason why, my cluster only provides gcc 7.3.0.
I'll try to get in touch with the administrators to see if there's something they can do.
I'll also keep an eye on your repo to see if you can figure out the issue with earlier gcc versions.

Thanks for your answer!

Best,
P

**Comment by francesco-peverelli** [comment]
Hi,

I have compiled with both gcc 7.3.0 and 6.3.0 without issue. It must be something else with your environment that is causing the error. Firstly, do use gcc 7.3.0 since I see from your error message that you are using 6.4.0 which I don't have available on my cluster, to eliminate that possible issue.
Secondly, I would make sure that you have pulled the latest commit and nothing is conflicting in your environment in terms of BMEAN and BMAN-GPU files (just to be sure). 
I have found a similar error on the NVIDIA forums, it seems to have something to do with some interaction of gcc with nvcc. According to this thread,
https://devtalk.nvidia.com/default/topic/1028112/cuda-setup-and-installation/nvcc-bug-related-to-gcc-6-lt-tuple-gt-header-/
the issue is fixed in CUDA 9.2. If you are able to upgrade to a later version (CUDA 10.x would be best) the issue may be resolved.
